splash/qtrender_lua.py

```diff
@@ -723,6 +737,38 @@ def set_header(self, name, value):
         self.request.setRawHeader(name, value)
 
 
+@contextlib.contextmanager
+def wrapped_response(lua, reply):
+    """
+    Context manager which returns a wrapped QNetworkReply
+    suitable for using in Lua code.
+    """
+    res = _WrappedResponse(lua, reply)
+    with lua.object_allowed(res, res.attr_whitelist):
+        yield res
+
+
+class _WrappedResponse(object):
+    _attribute_whitelist = ['commands', "response"]
+
+    def __init__(self, lua, response):
+        self.lua = lua
+        self.response = response
+        self.headers = dict([(str(k), str(v)) for k, v in response.rawHeaderPairs()])
```

Comment:
> While header->name mapping is convenient, it is not always correct 

this is valid point, but I see that PyQt4 QNetworkReply already handles this for us, if there are multiple headers with same key PyQT concatenates them together, you can test this like this

```python
def test_multiple_response_headers(self):
        resp = self.request_lua("""
        function main(splash)
            local h = nil
            splash:on_response_headers(function(response)
                h = response.get_headers()
            end)
            splash:http_get(splash.args.url)
            return h
        end
        """, {'url': "http://httpbin.org/response-headers?Set-Cookie=foo&Set-Cookie=bar&Set-Cookie=forest"})
```

above should return
```
{u'Access-Control-Allow-Credentials': u'true',
 u'Access-Control-Allow-Origin': u'*',
 u'Connection': u'keep-alive',
 u'Content-Length': u'130',
 u'Content-Type': u'application/json',
 u'Date': u'Tue, 07 Jul 2015 13:09:46 GMT',
 u'Server': u'nginx',
 u'Set-Cookie': u'foo\nbar\nforest'}
```
I see chrome devtools show multiple headers for this url:
```
Set-Cookie:foo
Set-Cookie:bar
Set-Cookie:forest
```

but they are concatenated by PyQt